restcomm/restcomm.mgcp/src/main/java/org/restcomm/connect/mgcp/IvrEndpoint.java

```diff
@@ -176,6 +175,28 @@ private void response(final Object message) {
         }
     }
 
+    private void handleAsrr(final Map<String, String> parameters, int code) {
+        if (parameters.containsKey("asrr")) {
+            String asrr = parameters.get("asrr");
+            if (!StringUtils.isEmpty(asrr)) {
+                try {
+                    asrr = new String(Hex.decodeHex(asrr.toCharArray()));
+                } catch (DecoderException e) {
+                    logger.error("asrr parameter cannot be decoded");
+                    fail(code);
+                }
+            }
+            // Notify the observers that the event successfully completed.
+            final IvrEndpointResponse result = new IvrEndpointResponse(new CollectedResult(asrr, true, code == 101));
+            for (final ActorRef observer : observers) {
+                observer.tell(result, self());
+            }
+        } else {
+            logger.error("asrr parameter is missing");
```

Comment:
can we add atleast code in this error msg 
or if it would make sense then print complete paramters